src/components/control-component.js

```diff
@@ -89,564 +89,578 @@ const propTypes = {
+      if (node) this.node = node;
+    }
 
-    if (node) this.node = node;
-  }
+    validate() {
+      const {
+        model,
+        modelValue,
+        fieldValue,
+        validators,
+        errors: errorValidators,
+        dispatch,
+      } = this.props;
 
-  validate() {
-    const {
-      model,
-      modelValue,
-      fieldValue,
-      validators,
-      errors: errorValidators,
-      dispatch,
-    } = this.props;
+      if (!validators && !errorValidators) return modelValue;
 
-    if (!validators && !errorValidators) return modelValue;
+      const fieldValidity = getValidity(validators, modelValue);
+      const fieldErrors = getValidity(errorValidators, modelValue);
 
-    const fieldValidity = getValidity(validators, modelValue);
-    const fieldErrors = getValidity(errorValidators, modelValue);
+      const errors = validators
+        ? merge(invertValidity(fieldValidity), fieldErrors)
+        : fieldErrors;
 
-    const errors = validators
-      ? merge(invertValidity(fieldValidity), fieldErrors)
-      : fieldErrors;
+      if (!shallowEqual(errors, fieldValue.errors)) {
+        dispatch(actions.setErrors(model, errors));
+      }
 
-    if (!shallowEqual(errors, fieldValue.errors)) {
-      dispatch(actions.setErrors(model, errors));
+      return modelValue;
     }
 
-    return modelValue;
-  }
-
-  render() {
-    const {
-      controlProps = emptyControlProps,
-      component,
-      control,
-    } = this.props;
-
-    const mappedProps = this.getMappedProps();
-
-    // If there is an existing control, clone it
-    if (control) {
-      return cloneElement(
+    render() {
+      const {
+        controlProps = emptyControlProps,
+        component,
         control,
+      } = this.props;
+
+      const mappedProps = this.getMappedProps();
+
+      // If there is an existing control, clone it
+      if (control) {
+        return cloneElement(
+          control,
+          {
+            ...mappedProps,
+            onKeyPress: this.handleKeyPress,
+          },
+          controlProps.children);
+      }
+
+      return createElement(
+        component,
         {
+          ...controlProps,
           ...mappedProps,
           onKeyPress: this.handleKeyPress,
         },
         controlProps.children);
     }
+  }
 
-    return createElement(
-      component,
-      {
-        ...controlProps,
-        ...mappedProps,
-        onKeyPress: this.handleKeyPress,
-      },
-      controlProps.children);
+  if (process.env.NODE_ENV !== 'production') {
+    Control.propTypes = propTypes;
   }
-}
 
-if (process.env.NODE_ENV !== 'production') {
-  Control.propTypes = propTypes;
+  Control.defaultProps = {
+    changeAction: actions.change,
+    updateOn: 'change',
+    asyncValidateOn: 'blur',
+    parser: identity,
+    controlProps: emptyControlProps,
+    getter: _get,
+    ignore: [],
+    dynamic: false,
+    mapProps: controlPropsMap.default,
+    component: 'input',
+    ...defaultProps,
```

Comment:
I added this so that these defaults can be overridden/appended to with any props passed in to the createControlClass() function